src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
 'use strict';
 
-// write code here
+// 1. Implement table sorting by clicking on the title (in two directions).
+const table = document.querySelector('table');
+
+function getTableRows() {
+  return [...table.querySelectorAll('tbody tr')];
+}
+
+function mapTableData(rows) {
+  return rows.map((row) => {
+    return [...row.children].map((cell) => cell.innerText);
+  });
+}
+
+function getTableData() {
+  const rows = getTableRows();
+
+  return mapTableData(rows);
+}
+
+let sortDirection = true;
+
+function sortTable(columnIndex) {
+  const sortedData = sortTableData(columnIndex, sortDirection);
+
+  renderSortedTable(sortedData);
+  toggleSortDirection();
+}
+
+function sortTableData(columnIndex, direction) {
+  const data = getTableData();
+
+  return data.sort((rowA, rowB) => {
+    let cellA = rowA[columnIndex];
+    let cellB = rowB[columnIndex];
+
+    cellA = parseCellData(cellA, columnIndex);
+    cellB = parseCellData(cellB, columnIndex);
+
+    if (direction) {
+      return cellA > cellB ? 1 : -1;
+    }
+
+    return cellA > cellB ? -1 : 1;
+  });
+}
+
+function parseCellData(cell, columnIndex) {
+  if (columnIndex === 3) {
+    return parseInt(cell);
+  }
+
+  if (columnIndex === 4) {
+    return parseFloat(cell.slice(1).replace(/,/g, ''));
+  }
+
+  return cell;
+}
```

Comment:
Code quality issue: The `parseCellData` function is tightly coupled with the current table structure. If the table structure changes, this function will break. Consider making this function more flexible, for example by using a configuration object that maps column indices to parsing functions.